main.py

```diff
@@ -409,6 +416,62 @@ def check_arg_name(args):
         )
 
 
+def reload_model(args, ckpt_path, model, index, strict, task):
+    """
+        Function that reloads model if necesary and extracts trainable part 
+        of the model in preparation for target_task training. 
+        It only reloads model after the first task is trained, and only 
+        for finetuning. 
+
+        Parameters
+        -------------------
+        args: config.Param object 
+        ckpt_path: str
+        model: MultiTaskmodel object 
+        index: int 
+        strict: bool, 
+        task: Task object 
+
+        Returns 
+        -------------------
+        to_train: List of tuples of (name, weight) for trainable parameters
+
+    """
+    if index > 0 and args.transfer_paradigm == "finetune":
+        # Only reload after target_train for first task, and transfer_paradigm=finetune.
+        load_model_state(model, ckpt_path, args.cuda, skip_task_models=[task.name], strict=strict)
+
+    if args.transfer_paradigm == "frozen":
```

Comment:
Does this really belong in reload_tasks? Seems like it'd make more sense to include in the setup fn.

Reply:
This is necessary in reload_tasks because every time you reload, you need to re-calculate which parameters to pass into the Trainer.